securedrop/server.py

```diff
@@ -177,6 +179,19 @@ def add_contact(self, email, contact_name, contact_email):
         self.write_json()
         return ""
 
+    def list_contacts(self, email):
+        if not email:
+            return "Invalid email address"
+        email_hash = hashlib.sha256((email.encode())).hexdigest()
+        users_keys_list = self.users.keys()
+        user = dict()
+
+        if email_hash in users_keys_list:
+            user = self.users[email_hash]
+            return user.contacts
+        else:
+            return user
```

Comment:
This is awkward. You don't need to check whether the email is in the dict, so you can just do:
```python
email_hash = hashlib.sha256((email.encode())).hexdigest()
return self.users[email_hash].contacts
```

Reply:
Replacing this block of code causes with the suggested causes an exception to thrown in server.py. I think the reason is that if "self.users[email_hash].contacts" is empty an empty dictionary is not returned.

Reply:
Bizarre... please look into this more. Are you sure that contacts is not `None`?

Reply:
I think contacts does evaluate to none. The problem is that this expression returns to the line below which in turn causes the exception.
"if email in self.email_to_sock and current_user_email in self.users.get_contacts(email):"

Reply:
I am pretty sure the problem is a KeyError:
https://realpython.com/python-keyerror/